### ISSUE TITLE
Soft reboot

### ISSUE THREAD

**Issue by caternuson** [adafruit/Adafruit_CircuitPython_IRRemote#3]
Getting this randomly. Not very frequent. Unfortunately can't correlate to anything. Happened at least once without any IR remote activity going on.
```
Traceback (most recent call last):
  File "main.py", line 11, in <module>
  File "adafruit_irremote.py", line 147, in read_pulses
KeyboardInterrupt: 
soft reboot

Auto-reload is on. Simply save files over USB to run them or enter REPL to disable.
main.py output:
```

**Comment by dhalbert** [comment]
This can happen due to delayed writes (nothing to do with the code). On Windows and Linux (and maybe even Mac sometimes), the main.py or whatever you're writing gets written, there's a soft reset, and then 20-90 seconds later some metadata gets written to the filesystem, and it soft-resets again.

If on a Unix-6 system, do a `sync` immediately after writing. If on Windows, do an Eject or the equivalent from the taskbar.

**Comment by caternuson** [comment]
This was on Ubuntu accessing the repl via screen.

Hmmmm. But I did have a file explorer window open to the CIRCUITPY folder. And maybe gedit. Wonder if that was maybe doing something?

**Comment by caternuson** [comment]
**TLDR**: It's most likely related to what @dhalbert said. Nothing to do with this lib so closing.

But, if you want to test, this seemed repeatable:
- Close any folder explorers or text editors you have open on the CP store
- Reset board
- Open a terminal and connect to board using screen
- Approx. 60secs later, get soft reboot.

Linux 4.10.0-35-generic #39~16.04.1-Ubuntu SMP Wed Sep 13 09:02:42 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
